util/options.py

```diff
@@ -258,26 +261,35 @@ def try_read_experiments_from_images(
         for arg in args:
             # Don't expand wildcards if URI-style filename
             if "*" in arg and not get_url_scheme(arg):
-                args_new.extend(glob(arg))
+                filenames = glob(arg)
+                if not filenames:
+                    logger.warning("No files found with pattern %s" % arg)
+                else:
+                    args_new.extend(glob(arg))
             else:
                 args_new.append(arg)
         args = args_new
 
         unhandled = []
-        experiments = ExperimentListFactory.from_filenames(
-            args,
-            unhandled=unhandled,
-            compare_beam=compare_beam,
-            compare_detector=compare_detector,
-            compare_goniometer=compare_goniometer,
-            scan_tolerance=scan_tolerance,
-            format_kwargs=format_kwargs,
-            load_models=load_models,
-        )
-        if experiments:
-            self.experiments.append(
-                FilenameDataWrapper(filename="<image files>", data=experiments)
+
+        try:
+            experiments = ExperimentListFactory.from_filenames(
+                args,
+                unhandled=unhandled,
+                compare_beam=compare_beam,
+                compare_detector=compare_detector,
+                compare_goniometer=compare_goniometer,
+                scan_tolerance=scan_tolerance,
+                format_kwargs=format_kwargs,
+                load_models=load_models,
             )
+            if experiments:
+                self.experiments.append(
+                    FilenameDataWrapper(filename="<image files>", data=experiments)
+                )
+        except FileNotFoundError as e:
+            logger.error("File %s not found", e.filename)
```

Comment:
Does the `FileNotFoundError` come from the `ExperimentListFactory.from_filenames` call or the `FilenameDataWrapper` call? If the former it might be slightly cleaner to do:
```suggestion

        try:
            experiments = ExperimentListFactory.from_filenames(
                args,
                unhandled=unhandled,
                compare_beam=compare_beam,
                compare_detector=compare_detector,
                compare_goniometer=compare_goniometer,
                scan_tolerance=scan_tolerance,
                format_kwargs=format_kwargs,
                load_models=load_models,
            )
        except FileNotFoundError as e:
            logger.error("File %s not found", e.filename)
        else:
            if experiments:
                self.experiments.append(
                    FilenameDataWrapper(filename="<image files>", data=experiments)
                )
```
Under what circumstances is a `FileNotFoundError` raise here?